Feature: Link Picker modal refactor

leekelleher opened the pull request:
## Description

A refactoring of the Link Picker modal UI.

Fixes and new features include...

- Sets the default modal size to "medium"
- Adds `hideTarget` flag
- Swaps the inline document & media trees with the corresponding input pickers (in their own modals)
- Populates the link title (if empty) with the selected document/media name
- Shows a media item preview; shows document `uui-ref-node` component
- Uses `<umb-property-layout>` component for UI consistency

Fixes https://github.com/umbraco/Umbraco-CMS/issues/16838
Fixes https://github.com/umbraco/Umbraco-CMS/issues/16596
Fixes https://github.com/umbraco/Umbraco-CMS/issues/17016


## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## How to test?

Test using either the Multi-URL Picker, TinyMCE or Tiptap editors. Can you add external links, pick a document or media item? Add, edit existing data, etc.

## Screenshots (if appropriate)

https://github.com/user-attachments/assets/b8fb201b-d506-4f38-af2d-7172c5db8508

madsrasmussen reviewed (changes requested):
It looks great!

I have a few minor UX tweaks for consistency. Firstly, we can remove the "Edit" action on the reference list items since we can already click on the name to achieve the same result. Additionally, we should follow the same convention inside the link picker modal when a document is selected. We should remove the "Open" action and make it possible to click the name.

leekelleher commented:
Thanks @madsrasmussen!

Changes made...

- Multi-URL Picker (`umb-input-multi-url`), removed the "Edit" action (as the `uui-ref-node` link title will open the edit link modal).
- Document Picker (`umb-input-document`), removed the "Open" action and added the feature to the `uui-ref-node` link title (to open the edit document workspace modal).
- In both components, hidden the `uui-action-bar` if in `readonly` mode.

madsrasmussen reviewed (approved):
🥳
